components/mpas-ocean/src/shared/mpas_ocn_time_average_coupled.F

```diff
@@ -584,4 +549,129 @@ subroutine ocn_time_average_coupled_accumulate(statePool, forcingPool, timeLevel
 
     end subroutine ocn_time_average_coupled_accumulate!}}}
 
+
+!***********************************************************************
+!
+!  routine accumulate_glc_3d_thermal_forcing
+!
+!> \brief   Coupled time averager accumulation for 3D thermal forcing
+!> \author  Xylar Asay-Davis
+!> \date    03/05/2025
+!> \details
+!>  This routine accumulated the coupled time average of the 3D thermal
+!>  forcing
+!
+!-----------------------------------------------------------------------
+    subroutine accumulate_glc_3d_thermal_forcing(statePool, forcingPool, timeLevel)!{{{
+        use ocn_constants, only: &
+             latent_heat_fusion_mks
+
+        type (mpas_pool_type), intent(in) :: statePool
+        type (mpas_pool_type), intent(inout) :: forcingPool
+        integer, intent(in) :: timeLevel
+
+        integer :: iCell
+        real (kind=RKIND), dimension(:,:), pointer :: avgThermalForcingAtZLevels
+        real (kind=RKIND), dimension(:,:), pointer :: avgThermalForcingAtZLevelsMask
+        real (kind=RKIND), dimension(:), pointer :: glcZLevels
+        real (kind=RKIND), dimension(:), pointer :: ssh
+
+        type (mpas_pool_type), pointer :: tracersPool
+
+        real (kind=RKIND), dimension(:,:,:), pointer :: activeTracers
+        integer, pointer :: indexTemperature, indexSalinity, nGlcZLevels, &
+                            nCells, nAccumulatedCoupled
+        integer :: iLevel, iLevelGlc, iLevelAbove, iLevelBelow
+
+        real (kind=RKIND) :: freezingTemp, glcz, zMin, zMax, zAbove, zBelow, &
+                             alpha, salin, temp, press
+
+        call mpas_pool_get_dimension(forcingPool, 'nCells', nCells)
+        call mpas_pool_get_dimension(forcingPool, 'nGlcZLevels', nGlcZLevels)
+
+        call mpas_pool_get_array(forcingPool, 'nAccumulatedCoupled', nAccumulatedCoupled)
+
+        call mpas_pool_get_array(forcingPool, 'glcZLevels', glcZLevels)
+        call mpas_pool_get_array(statePool, 'ssh', ssh, timeLevel)
+
+        call mpas_pool_get_array(forcingPool, 'avgThermalForcingAtZLevels', avgThermalForcingAtZLevels)
+        call mpas_pool_get_array(forcingPool, 'avgThermalForcingAtZLevelsMask', avgThermalForcingAtZLevelsMask)
+        call mpas_pool_get_subpool(statePool, 'tracers', tracersPool)
+        call mpas_pool_get_array(tracersPool, 'activeTracers', activeTracers, timeLevel)
+        call mpas_pool_get_dimension(tracersPool, 'index_temperature', indexTemperature)
+        call mpas_pool_get_dimension(tracersPool, 'index_salinity', indexSalinity)
+
+        !$omp parallel
+        !$omp do schedule(runtime)
```

Comment:
I think this is missing a bunch of `private()` variables that are causing very weird behavior.  Working on it...